packages/hw-app-xrp/src/Xrp.js

```diff
@@ -0,0 +1,180 @@
+ * const xrp = new Xrp(transport);
+ */
+export default class Xrp {
+  transport: Transport<*>;
+
+  constructor(transport: Transport<*>) {
+    this.transport = transport;
+    transport.setScrambleKey("w0w");
+  }
+
+  /**
+   * get Ripple address for a given BIP 32 path.
+   *
+   * @param path a path in BIP 32 format
+   * @param display optionally enable or not the display
+   * @param chainCode optionally enable or not the chainCode request
+   * @param ed25519 optionally enable or not the ed25519 curve (secp256k1 is default)
+   * @return an object with a publicKey, address and (optionally) chainCode
+   * @example
+   * const result = await xrp.getAddress("44'/144'/0'/0/0");
+   *
+   * {
+   *   "publicKey": "0324E5F600B52BB3D9246D49C4AB1722BA7F32B7A3E4F9F2B8A1A28B9118CC36C4",
+   *   "address": "rPwReL1dVYnVzFKShJhBxNPSbfX7rjJQEd"
+   * }
+   */
+  async getAddress(
+    path: string,
+    display?: boolean,
+    chainCode?: boolean,
+    ed25519?: boolean
+  ): Promise<{
+    publicKey: string,
+    address: string,
+    chainCode?: string
+  }> {
+    const bipPath = BIPPath.fromString(path).toPathArray();
+    const curveMask = ed25519 ? 0x80 : 0x40;
+
+    const cla = 0xe0;
+    const ins = 0x02;
+    const p1 = display ? 0x01 : 0x00;
+    const p2 = curveMask | (chainCode ? 0x01 : 0x00);
+    const data = Buffer.alloc(1 + bipPath.length * 4);
+
+    data.writeInt8(bipPath.length, 0);
+    bipPath.forEach((segment, index) => {
+      data.writeUInt32BE(segment, 1 + index * 4);
+    });
+
+    const response = await this.transport.send(cla, ins, p1, p2, data);
+
+    const result = {};
+    const publicKeyLength = response[0];
+    const addressLength = response[1 + publicKeyLength];
+
+    result.publicKey = response.slice(1, 1 + publicKeyLength).toString("hex");
+
+    result.address = response
+      .slice(1 + publicKeyLength + 1, 1 + publicKeyLength + 1 + addressLength)
+      .toString("ascii");
+
+    if (chainCode) {
+      result.chainCode = response
+        .slice(
+          1 + publicKeyLength + 1 + addressLength,
+          1 + publicKeyLength + 1 + addressLength + 32
+        )
+        .toString("hex");
+    }
+
+    return result;
+  }
+
+  /**
+   * sign a Ripple transaction with a given BIP 32 path
+   *
+   * @param path a path in BIP 32 format
+   * @param rawTxHex a raw transaction hex string
+   * @param ed25519 optionally enable or not the ed25519 curve (secp256k1 is default)
+   * @return a signature as hex string
+   * @example
+   * const signature = await xrp.signTransaction("44'/144'/0'/0/0", "12000022800000002400000002614000000001315D3468400000000000000C73210324E5F600B52BB3D9246D49C4AB1722BA7F32B7A3E4F9F2B8A1A28B9118CC36C48114F31B152151B6F42C1D61FE4139D34B424C8647D183142ECFC1831F6E979C6DA907E88B1CAD602DB59E2F");
+   *
+   * 30440220671dc22bed7006b2f170068ee0d9ddc5614b5fffdf438a69a956d9241c1d656b02207721ebf2d69c9c3938d6183bf087d10d786b9e94156ec46e4ddda9c8f7b95528
```

Comment:
same, probably that's ok not to show any result value. just in case someone use it